include/zephyr/net/wifi_credentials.h

```diff
@@ -56,6 +62,10 @@ struct wifi_credentials_header {
 	uint32_t timeout;                 /**< Timeout for connecting to the network */
 	uint8_t bssid[WIFI_MAC_ADDR_LEN]; /**< BSSID (Basic Service Set Identifier) */
 	uint8_t channel;                  /**< Channel on which the network operates */
+	char anon_id[16];                 /**< Anonymous identifier */
```

Comment:
IIRC, 16 was a hack that we did to WAR the length limitations of Wi-Fi credentials backends, can you please revisit this? At a minimum we need to document this limitation somewhere.

Reply:
For the time being, I have documented the `anon_id` limitation with the current value being used.